help: Improve groups documentation.

alya opened the pull request:
Update for recently added functionality.

- Lots of changes on https://zulip.com/help/manage-user-groups.
- The new intro also appears on https://zulip.com/help/user-groups.

- Updated https://zulip.com/help/create-user-groups
<details>
<summary>
Create user groups screenshot
</summary>

![Screenshot 2024-10-18 at 17 44 46@2x](https://github.com/user-attachments/assets/ec0c6a94-80f6-473f-bdfa-4c3218ddec3d)

</details>

I think we'll need additional documentation about the overall groups model, but this can be done as a follow-up.

alya commented:
@sahil839  Question I wasn't sure about: I think this needs to be updated? What exactly are the rules for what you do/don't need to be a group member in order to do?

![Screenshot 2024-10-18 at 17 30 07@2x](https://github.com/user-attachments/assets/606319f7-e587-43ba-92e7-3cac4e0c503b)

alya commented:
Also, @sahil839 , any feedback on the PR?

sahil839 commented:
Yes, I think that can be updated. We no longer have the requirement for non-admins and non-moderators to be member of the group to modify the user group. All the members of the group which is set to a setting are allowed now irrespective of their role. 

Maybe we can mention about guests, that guests do not have permissions to manage/modify (this will apply to all the administrative settings) the groups even if they are member of the group to which the setting is set.

sahil839 commented:
Changes look good overall.

alya commented:
Cool, I removed the old banner, and added this:
![Screenshot 2024-10-21 at 13 06 21@2x](https://github.com/user-attachments/assets/ad0be952-c394-440a-a45c-f1f214b2f014)

The point about guests was already explained in each org-level permissions sections.

alya commented:
@timabbott This PR was approved by @sahil839 .

timabbott commented:
Merged, thanks @alya and @sahil839!
